### PR TITLE
refactor: Replace lazy_static with std::sync::LazyLock

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Re-opening https://github.com/stackabletech/operator-rs/pull/827, after we got Rust 1.80 support via https://github.com/stackabletech/operator-templating/pull/420

Co-authored-by: Techassi <git@techassi.dev>

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Review by nightkr — DISMISSED** [review]

**Review by Techassi — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ env:
   CARGO_TERM_COLOR: always
   CARGO_INCREMENTAL: '0'
   CARGO_PROFILE_DEV_DEBUG: '0'
-  RUST_TOOLCHAIN_VERSION: "1.79.0"
```
Comment (Techassi) [inline .github/workflows/build.yml:20]: We should maybe bump to `1.80.1` here and below as well.

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.79.0"
```
Comment (Techassi) [inline rust-toolchain.toml:2]: Same bump to `1.80.1` possible here.

File: crates/k8s-version/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Changed
+
+- Replace `lazy_static` with `std::cell::LazyCell` ([#827]).
```
Comment (Techassi) [inline crates/k8s-version/CHANGELOG.md:9]: nit: I think we should also include the link to _this_ PR in all changelogs.

File: crates/stackable-operator/CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,8 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+- BREAKING: Replace `lazy_static` with `std::cell::LazyCell` (the original implementation was done in [#827] and reverted in [#835]) ([#XXX]).
```
Comment (Techassi) [inline crates/stackable-operator/CHANGELOG.md:7]: nit: Remove `xxx`.

```suggestion
- BREAKING: Replace `lazy_static` with `std::cell::LazyCell` (the original implementation was done in [#827] and reverted in [#835]) ([#840]).
```

**Review by Techassi — CHANGES_REQUESTED** [review]

File: crates/k8s-version/CHANGELOG.md (hunk 3 above)
Comment (sbernauer) [reply crates/k8s-version/CHANGELOG.md:9]: 98282dd

File: crates/stackable-operator/CHANGELOG.md (hunk 4 above)
Comment (sbernauer) [reply crates/stackable-operator/CHANGELOG.md]: 98282dd

File: .github/workflows/build.yml (hunk 1 above)
Comment (sbernauer) [reply .github/workflows/build.yml:20]: Good point, 7fe48c9

File: rust-toolchain.toml (hunk 2 above)
Comment (sbernauer) [reply rust-toolchain.toml:2]: Good point, 7fe48c9

File: crates/k8s-version/CHANGELOG.md (hunk 3 above)
Comment (Techassi) [reply crates/k8s-version/CHANGELOG.md:9]: `crates/k8s-version/CHANGELOG.md` still needs to mention _this_ PR.

File: rust-toolchain.toml (hunk 2 above)
Comment (Techassi) [reply rust-toolchain.toml:2]: I'm not quite sure why the commit is named _bump to 0.80.1_, but other than that, this comment is resolved.

Comment (sbernauer) [reply rust-toolchain.toml:2]: Hupsi, close enough ^^

File: crates/k8s-version/CHANGELOG.md (hunk 3 above)
Comment (sbernauer) [reply crates/k8s-version/CHANGELOG.md]: Good catch! 1e2cff3

**Review by Techassi — APPROVED** [review]

**Pull request merged**